nsxt/resource_nsxt_policy_lb_http_application_profile.go

```diff
@@ -0,0 +1,249 @@
+/* Copyright © 2020 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/bindings"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/data"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+var lBHttpProfileXForwardedForValues = []string{
+	model.LBHttpProfile_X_FORWARDED_FOR_REPLACE,
+	model.LBHttpProfile_X_FORWARDED_FOR_INSERT,
+}
+
+func resourceNsxtPolicyLBHttpApplicationProfile() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicyLBHttpApplicationProfileCreate,
+		Read:   resourceNsxtPolicyLBHttpApplicationProfileRead,
+		Update: resourceNsxtPolicyLBHttpApplicationProfileUpdate,
+		Delete: resourceNsxtPolicyLBHttpApplicationProfileDelete,
+		Importer: &schema.ResourceImporter{
+			State: nsxtPolicyPathResourceImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"nsx_id":       getNsxIDSchema(),
+			"path":         getPathSchema(),
+			"display_name": getDisplayNameSchema(),
+			"description":  getDescriptionSchema(),
+			"revision":     getRevisionSchema(),
+			"tag":          getTagsSchema(),
+			"http_redirect_to": {
+				Type:        schema.TypeString,
+				Description: "A URL that incoming requests for that virtual server can be temporarily redirected to, If a website is temporarily down or has moved",
+				Optional:    true,
+			},
+			"http_redirect_to_https": {
+				Type:        schema.TypeBool,
+				Description: "A boolean flag which reflects whether the client will automatically be redirected to use SSL",
+				Optional:    true,
+				Default:     false,
+			},
+			"idle_timeout": {
+				Type:         schema.TypeInt,
+				Description:  "Timeout in seconds to specify how long an HTTP application can remain idle",
+				Optional:     true,
+				Default:      15,
+				ValidateFunc: validation.IntAtLeast(1),
+			},
+			"ntlm": {
```

Comment:
This property is deprecated, judging by the text in docs - in this case, we shouldn't offer it in the provider